### PR TITLE
Fix disabled message.

### PR DESCRIPTION
**Pull request by blrubenstein**
I think that this fixed the all forbidden errors are user disabled piece? Also, this brings in the idea of an error code, which is a short message that describes the type of error.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/exceptions/ExceptionAdvice.java [hunk 1]
```diff
@@ -28,6 +28,10 @@
             ErrorResponse errorResponse = ((BadRequestException) e).getErrorResponse();
             return ResponseEntity.status(statusCode).body(
                 ((BadRequestException) e).getErrorResponse());
+          } else if (e instanceof ForbiddenException) {
+            ErrorResponse errorResponse = ((ForbiddenException) e).getErrorResponse();
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/exceptions/ExceptionAdvice.java]: opt: common base class or interface for these exceptions would likely be cleaner. Probably all of our exceptions setting an `ErrorResponse` (even without a specific code) would make sense, i.e. common base class.

File: ui/src/app/services/error-handling.service.ts [hunk 2]
```diff
@@ -73,7 +73,9 @@ export class ErrorHandlingService {
             this.setServerError();
             throw e;
           case 403:
-            this.setUserDisabledError();
+            if (JSON.parse(e._body).code === 'User Disabled') {
```
Comment (calbach) [inline ui/src/app/services/error-handling.service.ts]: Is `e._body` guaranteed to be present and non-empty? Possibly should guard against that being missing (unless this is guarantee by Angular or our swagger codegen).

File: api/src/main/java/org/pmiops/workbench/exceptions/ExceptionAdvice.java (hunk 1 above)
Comment (blrubenstein) [reply api/src/main/java/org/pmiops/workbench/exceptions/ExceptionAdvice.java]: Yeah, I can make a base class and just handle it all the same way.

File: api/src/main/resources/workbench.yaml [hunk 3]
```diff
@@ -1797,6 +1797,9 @@ definitions:
       message:
         description: General error message for the response.
         type: string
+      code:
```
[the remark this answers is not in the document]
Comment (blrubenstein) [reply api/src/main/resources/workbench.yaml]: Will do.

File: ui/src/app/services/error-handling.service.ts (hunk 2 above)
Comment (blrubenstein) [reply ui/src/app/services/error-handling.service.ts]: e._body will always be present _as long as we include any body in our errors._

I think any error we through should at least have a message included, so I feel pretty safe doing this, but if you would prefer to have a null check first, thats totally also doable.

Comment (calbach) [reply ui/src/app/services/error-handling.service.ts]: Thanks for checking on that. I'd prefer a null/undef check first in that case. This hooks into every outgoing XHR via the Angular stack, so in theory this error may not even have come from the Workbench API. There are also places in the Workbench API where we do things like forward errors from FC, where I'm less confident we'll always set an error body (though I think we probably are).

**Pull request merged**